src/BaGetter/ConfigureBaGetterServer.cs

```diff
@@ -0,0 +1,54 @@
+using BaGetter.Core;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Cors.Infrastructure;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.HttpOverrides;
+using Microsoft.Extensions.Options;
+
+namespace BaGetter;
+
+public class ConfigureBaGetterServer
+    : IConfigureOptions<CorsOptions>
+    , IConfigureOptions<FormOptions>
+    , IConfigureOptions<ForwardedHeadersOptions>
+    , IConfigureOptions<IISServerOptions>
+{
+    public const string CorsPolicy = "AllowAll";
+    private readonly BaGetterOptions _baGetterOptions;
+
+    public ConfigureBaGetterServer(IOptions<BaGetterOptions> baGetterOptions)
+    {
+        _baGetterOptions = baGetterOptions.Value;
+    }
+
+
+    public void Configure(CorsOptions options)
+    {
+        // TODO: Consider disabling this on production builds.
+        options.AddPolicy(
+            CorsPolicy,
+            builder => builder.AllowAnyOrigin()
+                .AllowAnyMethod()
+                .AllowAnyHeader());
+    }
```

Comment:
```suggestion
    private readonly BaGetterOptions _baGetterOptions;

    public ConfigureBaGetterServer(IOptions<BaGetterOptions> baGetterOptions, IWebHostEnvironment webHostEnvironment)
    {
        _webHostEnvironment = webHostEnvironment;
        _baGetterOptions = baGetterOptions.Value;
    }

    public void Configure(CorsOptions options)
    {
        if (!_webHostEnvironment.IsDevelopment()) return;

        options.AddPolicy(
            CorsPolicy,
            builder => builder.AllowAnyOrigin()
                .AllowAnyMethod()
                .AllowAnyHeader());
    }
```

Reply:
Yes it will: These `Allow` methods will be send as HTTP header from the server to the client's brower which then will relax its security policies. E.g: Allowing that javascripts can download resources from any website out there.

Reply:
If I change it to `if (!_webHostEnvironment.IsDevelopment()) return;` though, the CORS policy won't be set anymore in prod environments, so how will the HTTP headers be set then?

Reply:
> Yes it will: These `Allow` methods will be send as HTTP header from the server to the client's brower which then will relax its security policies. E.g: Allowing that javascripts can download resources from any website out there.

Also actually javascript can only download resources from any website out there if that web server sets its CORS headers to any.

Our settings set whether or not javascript from other websites can use endpoints of this BaGetter server.

Reply:
I think we can handle this TODO when someone has a proxy setup ready to test it.